src/core/tsi/ssl_transport_security.cc

```diff
@@ -2139,7 +2214,7 @@ tsi_result tsi_create_ssl_server_handshaker_factory_with_options(
         case TSI_REQUEST_AND_REQUIRE_CLIENT_CERTIFICATE_AND_VERIFY:
           SSL_CTX_set_verify(impl->ssl_contexts[i],
                              SSL_VERIFY_PEER | SSL_VERIFY_FAIL_IF_NO_PEER_CERT,
-                             nullptr);
+                             RootCertExtractCallback);
```

Comment:
I cann't comment right there but should l2207 also have this callback?

Reply:
Defer to @matthewstevenson88 on that, it's the case of `TSI_REQUEST_CLIENT_CERTIFICATE_AND_VERIFY`

Reply:
Yup good call, we should have it in both cases.

Reply:
Adding